libs/product/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -117,13 +121,13 @@ export const getDaffCompositeProductSelectors = (() => {
 		: createCompositeProductSelectors();
 })();
 
-function getOptionsDiscountAmount(product: DaffCompositeProduct, appliedOptions: Dictionary<string>): number {
+function getOptionsDiscountAmount(product: DaffCompositeProduct, appliedOptions: Dictionary<DaffCompositeProductEntityItem>): number {
 	return product.items.reduce((acc, item) => {
-		const itemOptionDiscount = item.options.find(option => option.id === appliedOptions[item.id]).discount;
+		const itemOptionDiscount = item.options.find(option => option.id === appliedOptions[item.id].value).discount;
 
 		return daffAdd(
 			acc, 
-			(itemOptionDiscount && itemOptionDiscount.amount > 0 ? itemOptionDiscount.amount : 0)
+			daffMultiply((itemOptionDiscount && itemOptionDiscount.amount > 0 ? itemOptionDiscount.amount : 0), appliedOptions[item.id].qty)
```

Comment:
I would do the `itemOptionDiscount && itemOptionDiscount.amount > 0` check outside of the `daffMultiply` call so that `daffMultiply` is only called when its needed.

Reply:
fixed